core/src/test/scala/quasar/plugin/postgres/PostgresDestinationSpec.scala

```diff
@@ -456,6 +456,62 @@ object PostgresDestinationSpec extends EffectfulQSpec[IO] with CsvSupport with P
       }
     }
 
+    "overwrite any existing table with the same name with WriteMode.Truncate" >>* {
+      csv(config(writeMode = WriteMode.Truncate)) { sink =>
+        val r1 = ("x" ->> 1) :: ("y" ->> "two") :: ("z" ->> 3.00001) :: HNil
+        val r2 = ("x" ->> 2) :: ("y" ->> "skldfj") :: ("z" ->> 3.00002) :: HNil
+
+        for {
+          tbl <- freshTableName
+          res1 <- drainAndSelect(TestConnectionUrl, tbl, sink, Stream(r1))
+          res2 <- drainAndSelect(TestConnectionUrl, tbl, sink, Stream(r2))
+        } yield (res1 must_=== List(r1)) and (res2 must_=== List(r2))
+      }
+    }
+
+    "append to existing table with WriteMode.Append" >>* {
```

Comment:
This test is failing, not sure why. `res2 must_=== List(r1) ++ List(r2)` is failing; it's returning just `List(r2)`.

Reply:
Ok, this test is failing against master. So either:
- appending is actual broken OR
- the test isn't testing what we think it is

Reply:
Ran the test against `master` branch. Went into postgres container.
```
postgres=# select * from "pgtest_test_YYux9O";
 x |   y    |    z
---+--------+---------
 2 | skldfj | 3.00002
(1 row)
```

`WriteMode.Append` appears borked!

Reply:
I suspect it's a problem with the test still though, or at least I haven't found anything wrong with the code. This would be easy to verify using Precog to run a full `Append` load.

Reply:
Ok, this has been resolved! @beckyconning noticed that the config was changing from `Append` to `Truncate` after saving a source with `Append`. Problem was a simple mis-mapping in the JSON codec.